### PR TITLE
[processing] restore simplify geometries & smooth algorithms

### PR DESCRIPTION
**Pull request by nirvn**
## Description
@nyalldawson , I was in need of the simplify geometries & smooth algorithms this morning. 

Porting went smoothly, nice job on making it as easy as possible while upgrading the whole processing platform.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/plugins/processing/algs/qgis/SimplifyGeometries.py [hunk 1]
```diff
@@ -58,20 +59,21 @@ def group(self):
...
-            self.methods, default=0))
-        self.addParameter(ParameterNumber(self.TOLERANCE,
-                                          self.tr('Tolerance'), 0.0, 10000000.0, 1.0))
+            self.methods, defaultValue=0))
+        self.addParameter(QgsProcessingParameterNumber(self.TOLERANCE,
+                                                       self.tr('Tolerance'), minValue=0.0, maxValue=10000000.0, defaultValue=1.0))
 
-        self.addOutput(OutputVector(self.OUTPUT, self.tr('Simplified')))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT, self.tr('Simplified')))
+        self.addOutput(QgsProcessingOutputVectorLayer(self.OUTPUT, self.tr('Simplified')))
```
Comment (nirvn) [inline python/plugins/processing/algs/qgis/SimplifyGeometries.py]: @nyalldawson , I'm curious, what's the difference between the addParameter's FeatureSink parameter vs. the addOutput? How are these two related to one another?

**Comment by nyalldawson** [comment]
Looks great, thanks! Can you also restore the tests for these algs?

File: python/plugins/processing/algs/qgis/SimplifyGeometries.py (hunk 1 above)
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/SimplifyGeometries.py]: Ok - so the feature sink parameter is the *input* specifying where the resulting layer should be created. That's the important bit.

Outputs are now used solely for indicating what generated values are available after running the alg within a model. So the algs need to report that they generate a vector layer, which can then be used as a input to other child algs later in a model.

There's a chance that some algs will create layers *without* giving users an option to specify where those layers are to be created, in which only the output will be set. This may be the case when a alg outputs multiple layers, or is hardcoded to place an output layer in a specific location. 

Similarly, there's a chance that algs will be written which generate layers, but (for whatever reason) don't want these layers available for use in models.

My intention here is that algs will be soaked in outputs. Things like outputing the count of features processed, etc. This would mean all those useful outputs become available for use in model logic.

**Comment by nyalldawson** [comment]
Sorry - one last thing. Can you add the feedback.isCanceled() check in the iterator loop too?

**Comment by nirvn** [comment]
@nyalldawson , done and done. BTW, I've renamed the smooth algorithm's INPUT_LAYER and OUTPUT_LAYER parameters to INPUT and OUTPUT. That's the standard you are pushing for right?

**Comment by nyalldawson** [comment]
> I've renamed the smooth algorithm's INPUT_LAYER and OUTPUT_LAYER parameters to INPUT and OUTPUT. That's the standard you are pushing for right?

Correct, thanks!

Sorry, I caused a merge conflict here. Can you rebase?

**Comment by nirvn** [comment]
@nyalldawson , _a_ conflict? You meant to say the second conflict today? 😉 No worries, I'll rebase.

File: python/plugins/processing/tests/testdata/qgis_algorithm_tests.yaml [hunk 2]
```diff
@@ -921,91 +921,91 @@ tests:
...
+      TOLERANCE: 1.0
+    results:
+      OUTPUT:
+        name: expected/simplify_multilines.gml
+        type: vector
+
+  - algorithm: qgis:simplifygeometries
+    name: Simplify (visval)
+    params:
+     INPUT:
```
Comment (nyalldawson) [inline python/plugins/processing/tests/testdata/qgis_algorithm_tests.yaml]: there's a missing space here throwing out travis

**Comment by nyalldawson** [comment]
Great work @nirvn!
